### PR TITLE
Modify destinations to avoid processing if they don't have settings

### PR DESCRIPTION
**Pull request by bsneed**
- Modified timeline processing such that we skip processing destinations that don't have settings.
- Added some convenience stuff necessary for tests.
- Added tests.

### CODE REVIEW

File: Sources/Segment/Settings.swift [hunk 1]
```diff
@@ -21,6 +21,15 @@ public struct Settings: Codable {
         ])
     }
     
+    public init(writeKey: String) {
```
Comment (prayansh) [inline Sources/Segment/Settings.swift:24]: ohh i might have missed this on kotlin 😅

File: Sources/Segment/Timeline.swift [hunk 2]
```diff
@@ -209,37 +209,40 @@ extension DestinationPlugin {
...
-            case let e as ScreenEvent:
-                destinationResult = screen(event: e) as? E
-            case let e as GroupEvent:
-                destinationResult = group(event: e) as? E
-            case let e as AliasEvent:
-                destinationResult = alias(event: e) as? E
-            default:
-                break
+
+        if let enabled = analytics?.settings()?.isDestinationEnabled(key: self.key), enabled == true {
```
Comment (prayansh) [inline Sources/Segment/Timeline.swift:218]: 👍

File: Sources/Segment/Timeline.swift [hunk 3]
```diff
@@ -209,37 +209,40 @@ extension DestinationPlugin {
...
+            
+            // apply .after plugins ...
+            result = timeline.applyPlugins(type: .after, event: destinationResult)
         }
         
-        // apply .after plugins ...
-        let afterResult = timeline.applyPlugins(type: .after, event: destinationResult)
-
-        return afterResult
+        return result
```
Comment (prayansh) [inline Sources/Segment/Timeline.swift:245]: so does it return `null` or `incomingEvent` if the destination is disabled?

**Review by prayansh — APPROVED** [review]
LGTM. just a question about the destination processing return value

File: Sources/Segment/Timeline.swift (hunk 3 above)
Comment (bsneed) [reply Sources/Segment/Timeline.swift:245]: it returns incomingEvent.  it ultimately ends up going into the ether, but it is consistent across the board.

**Pull request merged**